### PR TITLE
Restore statistical padding modes

### PR DESCRIPTION
**Pull request by fepegar**
[Generated by a coding agent]

---

## Summary

- restore `mean`, `median`, and `minimum` padding modes in `Pad` and `CropOrPad`
- preserve whole-volume, per-subject statistics across eager, batched, and lazy paths
- retain dtype, device, gradient, affine, and integer truncation behavior

## Testing

- `uv run --group test pytest tests/test_pad.py tests/test_crop_or_pad.py -q`
- `mise run quality`

### CODE REVIEW

File: src/torchio/transforms/spatial/pad.py [hunk 1]
```diff
@@ -33,6 +60,76 @@ def _parse_padding(padding: PaddingParam) -> TypeSixInts:
...
+        warnings.warn(
+            f'The constant value computed for padding mode "{padding_mode}"'
+            " might be truncated in the output, as the data type of the input"
+            " image is not float. Consider converting the image to a floating"
+            " point type before applying this transform.",
+            RuntimeWarning,
+            stacklevel=4,
+        )
+
+    float_flat = flat.float()
```
[the remark this answers is not in the document]
Comment (fepegar) [reply src/torchio/transforms/spatial/pad.py]: [Generated by a coding agent]

---

The calculation now preserves float32 and float64 inputs, while integer, float16, and bfloat16 inputs still use float32 accumulation. I also added exact regression coverage for float64 mean and median padding so this precision loss cannot recur.

https://github.com/TorchIO-project/torchio/blob/0b4378f58801b062d3f6de74b44f00df43dd9cc2/src/torchio/transforms/spatial/pad.py#L83-L90

Addressed in 0b4378f58801b062d3f6de74b44f00df43dd9cc2.

File: src/torchio/transforms/spatial/crop_or_pad.py [hunk 2]
```diff
@@ -452,7 +455,7 @@ def __init__(
             raise ValueError(msg)
         self.target_shape = _parse_target_shape(target_shape)
         self.units: Units = units
-        self.padding_mode = padding_mode
+        self.padding_mode = _parse_padding_mode(padding_mode)
```
[the remark this answers is not in the document]
Comment (fepegar) [reply src/torchio/transforms/spatial/crop_or_pad.py]: [Generated by a coding agent]

---

The public `Pad` and `CropOrPad` constructors now use `PaddingMode`, and I propagated that type through `EnsureShapeMultiple` and `GridSampler` so forwarded runtime values remain type-safe without weakening either transform annotation.

https://github.com/TorchIO-project/torchio/blob/167eaf7c920dd21e09986deadb5770dc04283e5a/src/torchio/transforms/spatial/crop_or_pad.py#L434-L445

Addressed in 167eaf7c920dd21e09986deadb5770dc04283e5a.

File: src/torchio/transforms/spatial/crop_or_pad.py [hunk 3]
```diff
@@ -29,6 +29,9 @@
 from ..transform import SpatialTransform
 from .crop import Crop
 from .pad import Pad
+from .pad import PaddingMode
+from .pad import _pad_tensor
+from .pad import _parse_padding_mode
```
[the remark this answers is not in the document]
Comment (fepegar) [reply src/torchio/transforms/spatial/crop_or_pad.py]: [Generated by a coding agent]

---

I moved the shared type, validation, statistic calculation, and tensor-padding implementation into the private `spatial._padding` module. `Pad` and `CropOrPad` now import intentional internal helpers (`parse_padding_mode` and `pad_tensor`) instead of reaching into each other's private implementation.

https://github.com/TorchIO-project/torchio/blob/1e46a258ae604aeed2d24c9e92550df00fbdc8da/src/torchio/transforms/spatial/_padding.py#L1-L35

Addressed in 1e46a258ae604aeed2d24c9e92550df00fbdc8da.

File: src/torchio/transforms/spatial/_padding.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+
+def pad_tensor(
+    data: Tensor,
+    padding: TypeSixInts,
+    padding_mode: PaddingMode,
+    fill: float,
+) -> Tensor:
+    """Pad a 4D image tensor or 5D image batch."""
+    assert data.ndim in (4, 5)
```
[the remark this answers is not in the document]
Comment (fepegar) [reply src/torchio/transforms/spatial/_padding.py]: [Generated by a coding agent]

---

The optimizable assertion has been replaced with an explicit `ValueError` that reports the received dimensionality. A direct regression test now covers the invalid-input path.

https://github.com/TorchIO-project/torchio/blob/11e0a4566bff1fcba6a1509670013114d104f535/src/torchio/transforms/spatial/_padding.py#L67-L76

Addressed in 11e0a4566bff1fcba6a1509670013114d104f535.

File: src/torchio/transforms/spatial/ensure_shape_multiple.py [hunk 5]
```diff
@@ -92,7 +94,7 @@ def __init__(
         target_multiple: TargetMultipleParam,
         *,
         method: str = "pad",
-        padding_mode: str = "constant",
+        padding_mode: PaddingMode = "constant",
         fill: float = 0,
```
[the remark this answers is not in the document]
Comment (fepegar) [reply src/torchio/transforms/spatial/ensure_shape_multiple.py:99]: [Generated by a coding agent]

---

`EnsureShapeMultiple` now validates and normalizes `padding_mode` during construction, matching `Pad` and `CropOrPad` fail-fast behavior. Invalid forwarded modes are covered by a constructor-level regression test.

https://github.com/TorchIO-project/torchio/blob/5cb47f48ec5c128e5aa6ede66ed417c3a117006b/src/torchio/transforms/spatial/ensure_shape_multiple.py#L93-L108

Addressed in 5cb47f48ec5c128e5aa6ede66ed417c3a117006b.

File: src/torchio/transforms/spatial/_padding.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+import warnings
+from typing import Literal
+from typing import get_args
+
+import torch
+from torch import Tensor
+
+from ...types import TypeSixInts
+from ..intensity.normalize import _quantile
```
[the remark this answers is not in the document]
Comment (fepegar) [reply src/torchio/transforms/spatial/_padding.py]: [Generated by a coding agent]

---

The large-tensor-safe quantile implementation now lives in the shared internal `transforms._statistics` module as `compute_quantile`. Both normalization and spatial padding import that stable internal helper, removing the spatial-to-intensity dependency and the private cross-subpackage import.

https://github.com/TorchIO-project/torchio/blob/79e565ce1184460e2bb7f33d102b51831c67c8ec/src/torchio/transforms/_statistics.py#L1-L25

Addressed in 79e565ce1184460e2bb7f33d102b51831c67c8ec.
